contracts/core/Migration.sol

```diff
@@ -3,25 +3,32 @@ pragma solidity ^0.8.19;
 
 import {IERC721Receiver} from "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
 import {INonfungiblePositionManager} from "../univ3-0.8/INonfungiblePositionManager.sol";
+import { IV3SwapRouter } from "../univ3-0.8/IV3SwapRouter.sol";
+import { ISwapRouter02 } from "../univ3-0.8/ISwapRouter02.sol";
+import { TransferHelper } from "../univ3-0.8/TransferHelper.sol";
 
 import "hardhat/console.sol";
 
 contract Migration is IERC721Receiver {
     INonfungiblePositionManager public immutable nfpm;
+    ISwapRouter02 public immutable swapRouter;
     address public immutable WEWE; 
     address public immutable WETH;
+    address public immutable USDC;
 
     struct PositionTokens {
         address token0;
         address token1;
     }
 
-    constructor(address _nfpm, address _WEWE, address _WETH) {
+    constructor(address _nfpm, address _swapRouter, address _WEWE, address _WETH, address _USDC) {
         require(_nfpm != address(0), "Migration: Invalid NonfungiblePositionManager address");
-
+        require(_nfpm != address(0), "_swapRouter: Invalid SwapRouter address");
+        swapRouter = ISwapRouter02(_swapRouter);
         nfpm = INonfungiblePositionManager(_nfpm);
         WEWE = _WEWE;
         WETH = _WETH;
+        USDC = _USDC;
     }
 
     function _decreaseAllLiquidity(uint256 tokenId) private returns (uint256 amount0, uint256 amount1) {
```

Comment:
If `_decreaseAllLiquidty(tokenId)` is not going to return any param, we can remove that returns... part